lib/aws_auth/authorization_header.ex

```diff
@@ -18,7 +18,11 @@ defmodule AWSAuth.AuthorizationHeader do
 
     headers = Map.put_new(headers, "host", uri.host)
 
-    payload = AWSAuth.Utils.hash_sha256(payload)
+    payload = if payload != :hashed do
```

Comment:
If with a negative condition and an else? Could you please make this a positive condition and flip the clauses.